Header File linking error with custom component

<!-- Thanks for reporting a bug for this project. READ THIS FIRST:
- Please make sure to submit issues in the right GitHub repository, if unsure just post it here:
  - esphomeyaml [here] - This is mostly for reporting bugs when compiling and when you get a long stack trace while compiling or if a configuration fails to validate.
  - esphomelib [https://github.com/OttoWinter/esphomelib] - Report bugs there if the ESP is crashing or a feature is not working as expected.
  - esphomedocs [https://github.com/OttoWinter/esphomedocs] - Report bugs there if the documentation is wrong/outdated.
- Provide as many details as possible. Paste logs, configuration sample and code into the backticks (```).
 
  DO NOT DELETE ANY TEXT from this template! Otherwise the issue may be closed without a comment.
-->

**Operating environment (Hass.io/Docker/pip/etc.):**
<!--

-->
Pi 3 HASSIO 0.86.0b0
**ESP (ESP32/ESP8266/Board/Sonoff):**
<!--

-->
Node MCU
**Affected component:**
<!--

-->
https://esphomelib.com/esphomeyaml/custom/uart.html

**Description of problem:**
Using the include in the yaml config file results in a unlinkable header in main.cpp.  Validate will pass but you can not compile.   

**Problem-relevant YAML-configuration entries:**
```yaml
esphomeyaml:
  name: alarm
  platform: ESP8266
  board: nodemcuv2
  includes:
   - my_custom_component.h

```

**Traceback (if applicable):**
<!--
Please copy the traceback here if compilation is failing. If possible, also connect to the ESP and copy its logs into the backticks.
-->
```
src/main.cpp:5:42: fatal error: ../../../my_custom_component.h: No such file or directory
#include "../../../my_custom_component.h"
^
compilation terminated.
*** [.pioenvs/alarm/src/main.cpp.o] Error 1
```

**Additional information:**
current work around, put the header file in
/config/esphomeyaml/alarm/.piolibdeps/esphomelib/src/esphomelib/my_custom_component.h
with alarm being my component name
and comment out the include statement in yaml file

--- a/esphomeyaml/core_config.py
+++ b/esphomeyaml/core_config.py
@@ -238,6 +238,6 @@ def includes(config):
     ret = []
     for include in config.get(CONF_INCLUDES, []):
         path = CORE.relative_path(include)
-        res = os.path.relpath(path, CORE.relative_build_path('src', 'main.cpp'))
+        res = os.path.relpath(path, CORE.relative_build_path('src'))
         ret.append(u'#include "{}"'.format(res))
     return ret

